### DISCUSSION TITLE
How to set platform to node for adapter-cloudflare-workers?

### DISCUSSION THREAD

**Discussion by HuakunShen** [sveltejs/kit#10731] [Q&A]
I am using @sveltejs/adapter-cloudflare-workers to build my sveltekit app but got the following error
I searched the documentation of sveltekit and cloudflare but couldn't find where I can set the platform.
The only related doc I found was https://developers.cloudflare.com/workers/runtime-apis/nodejs/. But setting this in `wrangler.toml` didn't help.

```
> Using @sveltejs/adapter-cloudflare-workers
✘ [ERROR] Could not resolve "crypto"

    ../../node_modules/@redis/client/dist/lib/lua-script.js:4:25:
      4 │ const crypto_1 = require("crypto");
        ╵                          ~~~~~~~~

  The package "crypto" wasn't found on the file system but is built into
  node. Are you trying to bundle for node? You can use "platform: 'node'"
  to do that, which will remove this error.

✘ [ERROR] Could not resolve "url"

    ../../node_modules/@redis/client/dist/lib/client/index.js:24:22:
      24 │ const url_1 = require("url");
         ╵                       ~~~~~

  The package "url" wasn't found on the file system but is built into
  node. Are you trying to bundle for node? You can use "platform: 'node'"
  to do that, which will remove this error.

✘ [ERROR] Could not resolve "util"

    ../../node_modules/@redis/client/dist/lib/cluster/cluster-slots.js:17:23:
      17 │ const util_1 = require("util");
         ╵                        ~~~~~~

  The package "util" wasn't found on the file system but is built into
  node. Are you trying to bundle for node? You can use "platform: 'node'"
  to do that, which will remove this error.

✘ [ERROR] Could not resolve "net"

    ../../node_modules/@redis/client/dist/lib/client/socket.js:16:20:
      16 │ const net = require("net");
         ╵                     ~~~~~

  The package "net" wasn't found on the file system but is built into
  node. Are you trying to bundle for node? You can use "platform: 'node'"
  to do that, which will remove this error.

✘ [ERROR] Could not resolve "tls"

    ../../node_modules/@redis/client/dist/lib/client/socket.js:17:20:
      17 │ const tls = require("tls");
         ╵                     ~~~~~

  The package "tls" wasn't found on the file system but is built into
  node. Are you trying to bundle for node? You can use "platform: 'node'"
  to do that, which will remove this error.
```

**Comment by jasongitmail** [comment]
From your link, it looks like the redis dependency might need to have its imports updated:

> Node.js APIs are available under the node: prefix, and this prefix must be used when importing modules, both in your code and the npm packages you depend on.
>
> // Do this:
> import { Buffer } from 'node:buffer';
> 
> // Do not do this:
> import { Buffer } from 'buffer';

I ran into something similar once, and the solution was to switch to a different dependency, for better or worse. It's Cloudflare related, not Svelte related.

If you can move services, you could try upstash, which offers serverless redis: https://upstash.com/

**Reply by HuakunShen** [reply]
Thanks I will jave a look.

**Reply by HuakunShen** [reply]
It looks like the `@upstash/redis` package only supports upslash? But I am using another redis service.

**Reply by jasongitmail** [reply]
That's why I said if you can move services. You'll likely have more luck asking your Redis provider directly, if you want to stick with them.
